maxim/dataset/dataset.py

```
@@ -109,3 +115,102 @@ def validate_data_structure(
+            converted_entries.append(entry)
+        elif isinstance(entry, dict):
+            # Convert dictionary to DatasetEntry using the new from_dict method
+            converted_entries.append(DatasetEntry.from_dict(entry))
+        else:
+            raise TypeError(f"Invalid entry type: {type(entry).__name__}. Expected DatasetEntry or dict.")
+
+    entries_with_row_no: List[DatasetEntryWithRowNo] = []
+    for i, entry in enumerate(converted_entries):
+        entries_with_row_no.extend(DatasetEntryWithRowNo.from_dataset_entry(entry, i + 1 + total_rows))
+
+    attachment_queue: List[DatasetEntryWithRowNo] = []
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)